src/app/(proper_react)/redesign/(authenticated)/user/welcome/EnterInfo.tsx

```diff
@@ -5,36 +5,79 @@
 "use client";
 
 import Image from "next/image";
-import { Session } from "next-auth";
+import { useRouter } from "next/navigation";
 import { FormEvent, useState } from "react";
+import { Session } from "next-auth";
 import { useOverlayTriggerState } from "react-stately";
 import { useOverlayTrigger } from "react-aria";
 import whyWeNeedInfoHero from "./images/welcome-why-we-need-info.svg";
-import { useL10n } from "../../../../hooks/l10n";
-import { ModalOverlay } from "../../../../components/client/dialog/ModalOverlay";
-import { Dialog } from "../../../../components/client/dialog/Dialog";
-import { Button } from "../../../../components/server/Button";
-import { InputField } from "../../../../components/client/InputField";
-import { LocationAutocompleteInput } from "../../../../components/client/LocationAutocompleteInput";
+import { useL10n } from "../../../../../hooks/l10n";
+import { ModalOverlay } from "../../../../../components/client/dialog/ModalOverlay";
+import { Dialog } from "../../../../../components/client/dialog/Dialog";
+import { Button } from "../../../../../components/server/Button";
+import { InputField } from "../../../../../components/client/InputField";
+import {
+  LocationAutocompleteInput,
+  getDetailsFromLocationString,
+} from "../../../../../components/client/LocationAutocompleteInput";
+import { WelcomeScanBody } from "../../../../../api/v1/user/welcome-scan/create/route";
+import { StateAbbr } from "../../../../../../utils/states";
+import { ISO8601DateString } from "../../../../../../utils/parse.js";
 
 import styles from "./EnterInfo.module.scss";
 
+export type UserInfo = {
+  firstName: string;
+  lastName: string;
+  city: string;
+  state: StateAbbr;
+  dateOfBirth: ISO8601DateString;
+};
+
+const getAgeFromDateString = (dateOfBirth: ISO8601DateString): number => {
+  const dateNow = new Date();
+  const dateBirth = new Date(dateOfBirth);
+  const dateDelta = new Date(dateNow.valueOf() - dateBirth.valueOf());
+  const unixStartDate = new Date(0);
+
+  return dateDelta.getUTCFullYear() - unixStartDate.getUTCFullYear();
+};
```

Comment:
Not sure if this is the most bulletproof way to check the age from a date, but I think it is working. This is only for validating the field input and that OneRep will reject a profile that has a birth date under 13. Should we add additional validation to our `/api/v1/user/welcome-scan/create` route @rhelmer @mansaj?

Reply:
Yes I think we should guard against this on the route as well.

Reply:
I'd suggest moving this to a shared function so we can use it from the route as well. I think this approach is OK (the new Temporal API or a third-party library like `moment` is easier but I think sticking to standard JS for now is good).

As we already discussed maybe `isEligible()` function could be the function that we call, currently it just checks that the user has not already performed a scan, but checking that they are of the correct age would be useful too.

Reply:
Moved the age check into a shared function and am also calling it now additionally in the `/welcome-scan/create` route: [31a6f65](https://github.com/mozilla/blurts-server/pull/3222/commits/31a6f650df5673b5598881022f875d0ea625d83b).